homeassistant/components/tradfri.py

```diff
@@ -58,13 +57,27 @@ def configuration_callback(callback_data):
         """Handle the submitted configuration."""
         try:
             from pytradfri.api.aiocoap_api import APIFactory
+            from pytradfri import RequestError
+            from uuid import uuid4
         except ImportError:
             _LOGGER.exception("Looks like something isn't installed!")
             return
 
-        api_factory = APIFactory(host, psk_id=GATEWAY_IDENTITY)
-        psk = yield from api_factory.generate_psk(callback_data.get('key'))
-        res = yield from _setup_gateway(hass, config, host, psk,
+        identity = uuid4().hex
+        security_code = callback_data.get('security_code')
+
+        api_factory = APIFactory(host, psk_id=identity, loop=hass.loop)
+        # Need To Fix: currently entering a wrong security code sends
```

Comment:
Is this existing behavior or is this something that this PR introduces?

Reply:
that is existing behaviour, or at least something that originates from the error handing in that tdls transport module in aiocoaps.  It should be something that gets fixed in aiocoap or pytradfri. There's really nothing we can do to handle/catch it here for as far as we could see.

I'm personally looking into it too, but my (async) python skills are still 'evolving' ;)

Reply:
I created an issue on this here: https://github.com/chrysn/aiocoap/issues/96